contracts/src/v0.8/feeds/DualAggregator.sol

```diff
@@ -538,32 +541,65 @@ contract DualAggregator is OCR2Abstract, OwnerIsCreator, AggregatorV2V3Interface
     return s_hotVars.latestSecondaryRoundId;
   }
 
+  /**
+   * @notice check if a report has already been transmitted
+   * @param report the report to check
+   */
+  function _reportExists(
+    Report memory report
+  ) internal view returns (bool exist, uint32 roundId) {
+    // get the latest round id
+    uint32 latestAggregatorRoundId = s_hotVars.latestAggregatorRoundId;
+
+    for (uint32 round_ = latestAggregatorRoundId; round_ > 0; --round_) {
```

Comment:
We likely need to cap this with a MAX_ITERATIONS constant.

Reply:
I expected the first loop check to be the cap mechanism:
```
if (transmission.observationsTimestamp < report.observationsTimestamp) {
    return (false, 0);
  }
```
We could add a MAX_ITERATIONS too, but are we sure to want it as a constant? I believe a variable can be a better option if we want to change the MAX_ITERATIONS in the future, specially because I expect us to have blockchains with long heartbeats and blockchains with instant heartbeats.

Reply:
In the case the secondary feed is down for very long this could iterate for a while. A variable works too.

Reply:
Done